File: Data Structures & Algorithms/two-integer-sum/submission-3.go
func twoSum(nums []int, target int) []int {

    c := make(map[int]int)

    c[target - nums[0]] = 0

    for i:=1; i < len(nums); i++ {
        if f, found := c[nums[i]]; found {
            return []int{f, i}
        }
        c[target-nums[i]] = i
    }
    return []int{0, 1}
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.kt
class Solution {
    fun twoSum(numbers: IntArray, target: Int): IntArray {
        var l = 0
        var r = numbers.size - 1

        while (l < r) {
            val sum = numbers[l] + numbers[r]
            if (sum == target) break
            if (sum > target) {
                r--
            } else {
                l++
            }
        }

        return intArrayOf(l+1,r+1)
    }
}
